include/IECoreGL/Texture.h

```diff
@@ -96,6 +96,10 @@ class IECOREGL_API Texture : public Bindable
 		/// Derived classes must set this in their constructor.
 		GLuint m_texture;
 
+		// \todo : Needed by hack in Shader::Setup::MemberData::TextureValue until we are able
+		// to break ABI and add targetType support to Texture
+		friend class Shader;
```

Comment:
At some point I think we might as well add a public accessor for the texture ID anyway.